File: go/two-fer/two_fer.go
// Package twofer implements a function to help determine what you
// will say when handing out cookies to friends
package twofer

import "fmt"

// ShareWith accepts the param "name" as a string and returns
// the string "One for you, one for me." if the name is ""
// or it returns the name in place of "you"
func ShareWith(name string) string {
	if name == "" {
		return "One for you, one for me."
	}

	return fmt.Sprintf("One for %s, one for me.", name)
}

File: elixir/two-fer/lib/two_fer.ex
defmodule TwoFer do
  @moduledoc """
  TwoFer contains the two_fer function, which returns the correct string for the given input
  """
  @doc """
  Two-fer or 2-fer is short for two for one. One for you and one for me.
  """
  @spec two_fer(name :: String.t()) :: String.t()
  def two_fer(name \\ "you") when is_binary(name), do: "One for #{name}, one for me."
end
